engine/bsql_engine/io/cio.pxd

```diff
@@ -109,7 +109,7 @@ cdef extern from "../include/io/io.h":
     pair[bool, string] registerFileSystemGCS( GCS gcs, string root, string authority) except +raiseRegisterFileSystemGCSError
     pair[bool, string] registerFileSystemS3( S3 s3, string root, string authority) except +raiseRegisterFileSystemS3Error
     pair[bool, string] registerFileSystemLocal(  string root, string authority) except +raiseRegisterFileSystemLocalError
-    TableSchema parseSchema(vector[string] files, string file_format_hint, vector[string] arg_keys, vector[string] arg_values, vector[pair[string,type_id]] types) except +raiseParseSchemaError
+    TableSchema parseSchema(vector[string] files, string file_format_hint, vector[string] arg_keys, vector[string] arg_values, vector[pair[string,type_id]] types, bool ignore_missing_paths) except +raiseParseSchemaError
```

Comment:
why is this a configuration. Who sets this ignore_missing_paths?

Reply:
The ignore_missing_paths feature was added because we are wanting to allow a user to specify  a set of partitions for a table. This means we need to constructed the expected paths, but not all partition combinations may exist, which means we need to be ok with contructing a path that may not actually exist, and therefore not really be an error. For example:
Say the user requests:  partitions:{'colA' :[1, 2], 'colB':[3, 4]}
So from this we build four directories:
base_folder/colA=1/colB=3/*
base_folder/colA=1/colB=4/*
base_folder/colA=2/colB=3/*
base_folder/colA=2/colB=4/*
But if the partition combination colA=1/colB=4 did not actually exist, then that path does not exist and its ok. We dont want to throw an error. In order to enable that I introduced that ignore_missing_paths functionality. It only ever gets used during `parseSchema` and only if the user is specifying partitions without using Hive.

Reply:
ok I see so the actual folders coudl be missing and it would throw an error. Yah I cant think of another way around this. What about the case when the user fucks up and NONE of the path exist? Or like I specify

col a = [part1] col b [part2] and there are NO values for anyone in part1 so it would be empty, do we want to let them konw that?, or even worse what if the user specifies col abc when no column abc exists?

Reply:
Ok, so if the user screwed up and like col abc does not exist, then it will find no folders and no files when trying to get the schema, in which case it should throw an error there. I will test out that scenario and make sure it works